sycl/source/handler.cpp

```diff
@@ -253,10 +253,8 @@ event handler::finalize() {
     if (MQueue && !impl->MGraph && !impl->MSubgraphNode &&
         !MQueue->getCommandGraph() && !impl->CGData.MRequirements.size() &&
         !MStreamStorage.size() &&
-        (!impl->CGData.MEvents.size() ||
-         (MQueue->isInOrder() &&
-          detail::Scheduler::areEventsSafeForSchedulerBypass(
-              impl->CGData.MEvents, MQueue->getContextImplPtr())))) {
+        detail::Scheduler::areEventsSafeForSchedulerBypass(
+            impl->CGData.MEvents, MQueue->getContextImplPtr())) {
```

Comment:
Side note, would it be possible to change `areEventsSafeForSchedulerBypass` to taking a `const ContextImplPtr&` instead? Seems like we might as well save the `shared_ptr` copy unless strictly needed.